.github/workflows/build.yml

```diff
@@ -2,9 +2,8 @@ on:
   push:
     branches:
       - master
-  pull_request:
-    branches:
-      - master
+  pull_request_target:
+    types: [opened, closed, reopened]
```

Comment:
I don't see any reason to trigger on the `closed` event here. You probably also want to add `synchronize` - it's poorly named but it's triggered:

> when the pull request’s tracking branch is synchronized with the source branch for the pull request, and that happens when the source branch is updated.

[Source](https://github.community/t/what-is-a-pull-request-synchronize-event/14784)

Reply:
Done!